### DISCUSSION TITLE
Mule broken on EX (Gen1a) after latest scitools update

### DISCUSSION THREAD

**Discussion by epitome76** [MetOffice/simulation-systems#277] [Mule]
Hi, I use mule in a script and start with pre-script='module load scitools/default; module load um_tools; mule-version'. On EX this started failing yesterday. It still works with default-previous, so it must be something to do with scitools? The error is:

```
  File "/common/internal/umdir/mule/mule-2023.08.1/python-3.6.15_numpy-1.17.3/openmp/lib/mule/packing.py", line 107, in <module>
    raise ImportError(err.args + (msg,))
ImportError: ("No module named 'imp'", 'SHUMlib Packing library found, but failed to import')

```
It seems to work on XC40, so it might be a EX specfic thing, let me know if this is better brought up with the porting people.

**Comment by james-bruten-mo** [comment]
Hi, the default version of scitools changed yesterday which is why you're seeing this now. The latest versions of scitools-default-current and default-next use a python version that's incompatible with mule 2023.08.1. We're in the process of installing mule 2024.11.1 which works with the later python versions, so I'd suggest switching to using that.

It's also worth considering explicitly loading a module version so you have a more stable environment. I'd suggest something like `module load scitools/default-2024_11_26 ; module load um_tools/2024.11.1/openmp` should work when the latest um_tools is ready.

**Reply by epitome76** [reply]
Thanks, James. It's good to know you are aware of this and have a solution. I will implement your suggestion.
